### PR TITLE
HSV should wrap hue values % 360 otherwise it outputs an invalid hue of black

### PR DESCRIPTION
**Pull request by NigelBreslaw**

### CODE REVIEW

File: .github/ci_path_filters.yaml [hunk 1]
```diff
@@ -21,7 +21,8 @@ slint:
  - 'logo/**'
  - 'scripts/**'
  - 'tests/**'
- - 'tools/(!figma-inspector/)**'
+ - '!tools/figma-inspector/**'
+ - 'tools/**'
```
Comment (ogoffart) [inline .github/ci_path_filters.yaml]: i hope this doesn't make all the test run when only files in the figma-inspector are changed.

**Review by ogoffart — APPROVED** [review]
Could you also add a test for negative value of the hue?

File: tests/cases/types/color.slint [hunk 2]
```diff
@@ -59,6 +59,8 @@ Test := Rectangle {
 
     out property <bool> test_hsv: gr1.to-hsv().hue == new_green.to-hsv().hue && gr1.to-hsv().saturation == new_green.to-hsv().saturation && gr1.to-hsv().value == new_green.to-hsv().value
         && bwood == Colors.hsv(bwood_hue, bwood_sat, bwood_bri);
+    out property <bool> test_hsv_wrap: gr1.to-hsv().hue == new_green.to-hsv().hue && gr1.to-hsv().saturation == new_green.to-hsv().saturation && gr1.to-hsv().value == new_green.to-hsv().value
```
Comment (ogoffart) [inline tests/cases/types/color.slint:62]: You also need to add this test to the `test` condition.
(ie: add a ` && test_hsv_wrap` a few line down.

File: .github/ci_path_filters.yaml (hunk 1 above)
Comment (NigelBreslaw) [reply .github/ci_path_filters.yaml]: I removed this. @tronical was already properly fixing the CI

File: tests/cases/types/color.slint (hunk 2 above)
Comment (NigelBreslaw) [reply tests/cases/types/color.slint:62]: Thanks!

File: tests/cases/types/color.slint [hunk 3]
```diff
@@ -59,12 +59,13 @@ Test := Rectangle {
 
     out property <bool> test_hsv: gr1.to-hsv().hue == new_green.to-hsv().hue && gr1.to-hsv().saturation == new_green.to-hsv().saturation && gr1.to-hsv().value == new_green.to-hsv().value
         && bwood == Colors.hsv(bwood_hue, bwood_sat, bwood_bri);
+    out property <bool> test_hsv_wrap: Colors.blue == hsv(240, 1, 1) && Colors.blue == hsv(600, 1, 1) && Colors.blue == hsv(-120, 1, 1);
```
Comment (tronical) [inline tests/cases/types/color.slint:62]: `Colors.blue == hsv(600, 1, 1)`

This fails because `hsv()` is documented to take a value between 0 and 360 for the hue. Not documented, but implemented, is that values outside are clamped.

So it would be kind of weird if the Rust HSV functions wraps and the Slint function clamps. We need to agree and sync them.

Technically, one could argue that changing from clamp to wrap is breaking. That said, if we can do the wrap here in one place, then the generated code becomes lighter. That's attractive.

@ogoffart thoughts?

Comment (NigelBreslaw) [reply tests/cases/types/color.slint:62]: Due to this issue any hue of 360 and greater results in an output of black. So there are kind of 2 bugs. The docs say 360 is valid, but it isn't. 359.99999 is the last valid number I think? Then black is not even a hue. I can imagine though that someone has code that depends on this as their odd way of creating something black. However I would like to think this would be super rare and it won't be hard to spot when suddenly things work correctly?

Comment (tronical) [reply tests/cases/types/color.slint:62]: To add what Nigel wrote (which I find compelling): It appears the the interpreter in `eval.rs` does *not* clamp the hue.

Comment (ogoffart) [reply tests/cases/types/color.slint:62]: IMHO we should do whatever CSS does 
the hue should be mod 360 because it is an angle and that wraps,  the other value should be clamped in their valid range.

**Review by tronical — APPROVED** [review]
